### PR TITLE
KTOR-2816 Adding application startup and hot-reloading time log

### PR DESCRIPTION
**Pull request by BOOMeranGG**
**Subsystem**
Server - core

**Motivation**
I saw a issue #2006 and that sounds good for me too.
There is an open pull requests from Mar/2021 #2407 by @jgabrielfreitas, but the issue has not been resolved since. I think this task is perfect as a first contribution to open source.

**Solution**
The solution has 2 types of logs:
- For the first loading calculating time between start of `init` and receiving the `ApplicationStarted` event.
- For the auto-reload calculating time between receiving `ApplicationStarting` and `ApplicationStarted` events.

### CODE REVIEW

File: ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/BaseApplicationEngine.kt [hunk 1]
```diff
@@ -30,21 +30,36 @@ public abstract class BaseApplicationEngine(
...
+            val finishedAt = System.currentTimeMillis()
             environment.connectors.forEach {
                 environment.log.info(
                     "Responding at ${it.type.name.lowercase(Locale.getDefault())}://${it.host}:${it.port}"
                 )
             }
+
+            val elapsedTimeInSeconds = (finishedAt - initializedStartAt) / 1_000.0
+            if (isFirstLoading) {
+                environment.log.info("Started application in $elapsedTimeInSeconds seconds.")
```
Comment (rsinukov) [inline ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/BaseApplicationEngine.kt:58]: I believe `Application started in` sounds more natural than `Started application in`

**Review by rsinukov — CHANGES_REQUESTED** [review]
Thanks for your contribution. Looks good, but minor comments.

**Comment by BOOMeranGG** [comment]
Thanks for the comments, changes have been pushed

File: ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/BaseApplicationEngine.kt [hunk 2]
```diff
@@ -31,20 +31,34 @@ public abstract class BaseApplicationEngine(
...
                     "Responding at ${it.type.name.lowercase(Locale.getDefault())}://${it.host}:${it.port}"
                 )
             }
+
+            val elapsedTimeInSeconds = (finishedAt - initializedStartAt) / 1_000.0
+            if (isFirstLoading) {
+                environment.log.info("Application started in $elapsedTimeInSeconds seconds.")
+                isFirstLoading = false
+            } else {
+                environment.log.info("Auto-reload application in $elapsedTimeInSeconds seconds.")
```
Comment (rsinukov) [inline ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/BaseApplicationEngine.kt:60]: Sorry, I forgot to add a comment in the last review. Same as in the previous log message, `Application auto-reloaded`

**Review by rsinukov — CHANGES_REQUESTED** [review]

**Comment by rsinukov** [comment]
Also, please create a ticket in the [youtrack](https://youtrack.jetbrains.com/newIssue?project=KTOR) and mention it in the PR name.

File: ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/BaseApplicationEngine.kt (hunk 2 above)
Comment (BOOMeranGG) [reply ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/BaseApplicationEngine.kt]: Exactly, I should have guessed about it myself; /)

**Pull request merged**